genie-agent/src/main/java/com/netflix/genie/agent/execution/services/impl/FetchingCacheServiceImpl.java

```diff
@@ -140,7 +140,13 @@ private void lookupOrDownload(
         final String resourceCacheId = getResourceCacheId(sourceFileUri);
 
         // Get a handle to the resource
-        final Resource resource = resourceLoader.getResource(uriString);
+        final Resource resource;
+        try {
+             resource = resourceLoader.getResource(uriString);
+        } catch (Throwable t) {
+            log.error("Failed to retrieve resource: {}", uriString);
```

Comment:
are you purposely not adding `t` to this to avoid publishing the stack trace?

Reply:
Yes i just wanted to print what is failing to download.
This was motivated by analyzing Terra's problem the other day.